packages/core/admin/server/services/api-token.js

```diff
@@ -74,20 +74,13 @@ const create = async attributes => {
 /**
  * @returns {void}
  */
-const createSaltIfNotDefined = () => {
-  if (strapi.config.get('admin.apiToken.salt')) {
-    return;
-  }
-
-  if (process.env.API_TOKEN_SALT) {
+const checkSaltIsDefined = () => {
+  if (!strapi.config.get('admin.apiToken.salt')) {
+    const secretExample = crypto.randomBytes(16).toString('base64');
     throw new Error(
-      `There's something wrong with the configuration of your api-token salt. If you have changed the env variable used in the configuration file, please verify that you have created and set the variable in your .env file.`
+      `Missing apiToken.salt. Please set apiToken.salt in config/admin.js (ex: ${secretExample})`
```

Comment:
We shouldn't tell them to statically set the key in this file, they should use the environment variable.